include/net/net_l2.h

```diff
@@ -104,9 +104,15 @@ NET_L2_DECLARE_PUBLIC(BLUETOOTH_L2);
 NET_L2_DECLARE_PUBLIC(OPENTHREAD_L2);
 #endif /* CONFIG_NET_L2_OPENTHREAD */
 
+#ifdef CONFIG_NET_L2_CANBUS_RAW
+#define CANBUS_RAW_L2		CANBUS_RAW
+#define CANBUS_RAW_L2_CTX_TYPE	void*
+NET_L2_DECLARE_PUBLIC(CANBUS_L2_RAW);
+#endif /* CONFIG_NET_L2_CANBUS_RAW */
+
 #ifdef CONFIG_NET_L2_CANBUS
 #define CANBUS_L2		CANBUS
-#define CANBUS_L2_CTX_TYPE	void*
+#define CANBUS_L2_CTX_TYPE	struct net_can_context *
```

Comment:
I recommend putting `#define CANBUS_L2_CTX_TYPE	struct net_can_context *` into canbus related header file.

Reply:
Isn't net_l2.h the right point for that? All other L2 CTX types are defined here. I don't like to include internal headers in other files using this definitions.

Reply:
All the L2 CTX types that are defined in net_l2.h have a type "void *". But all the other L2 technologies that are using a real type, have their L2 CTX defined in relevant header file (for example ethernet.h and new ppp support that is coming). The benefit of this (to put the real context type into somewhere else) is that if you enable CONFIG_NET_L2_CANBUS, you need to include can.h in certain order so that compile will not complain about missing "struct net_can_context".

Reply:
Moved it to include/net/can.h